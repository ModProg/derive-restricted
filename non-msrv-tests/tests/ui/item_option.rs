#[derive(derive_where::DeriveWhere)]
#[derive_where(Debug(option); T)]
struct Test1<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Clone; T)]
#[derive_where(skip_inner)]
enum Test2<T> {
	A(T),
}

#[derive(derive_where::DeriveWhere)]
#[derive_where(Debug; T)]
#[derive_where(skip_inner)]
#[derive_where(skip_inner)]
struct Test3<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Debug; T)]
#[derive_where(skip_inner())]
struct Test4<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Debug; T)]
#[derive_where(skip_inner)]
#[derive_where(skip_inner(Debug))]
struct Test5<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Clone; T)]
#[derive_where(skip_inner)]
struct Test6<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Clone; T)]
#[derive_where(skip_inner(Clone))]
struct Test7<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Debug; T)]
#[derive_where(skip_inner(Debug, Debug))]
struct Test8<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Debug; T)]
#[derive_where(skip_inner(Debug))]
#[derive_where(skip_inner(Debug))]
struct Test9<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Clone; T)]
#[derive_where(skip_inner(Debug))]
struct Test10<T>(T);

fn main() {}
