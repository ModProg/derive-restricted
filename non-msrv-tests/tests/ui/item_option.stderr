error: `Debug` doesn't support any options
 --> tests/ui/item_option.rs:2:16
  |
2 | #[derive_where(Debug(option); T)]
  |                ^^^^^^^^^^^^^
